### PR TITLE
RFC: Fetch and merge upstream changes

### PR DESCRIPTION
**Pull request by falkTX**
Brings changes from upstream master branch.
README.md and 3RDPARTY.txt files had conflicts.
README.md conflicts were easily resolved.
3RDPARTY.txt file is not present upstream, do we need it? Or delete it as well? (which I did here, not sure if correct)

- [x] The code in this pull request is licensed under "GPLv2 or any later version"
- [ ] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"

### CODE REVIEW

**Comment by falkTX** [comment]
Note: in order to keep proper history if these PRs are merged, they need to be in "rebase and merge" mode.

**Comment by Semisol** [comment]
> Note: in order to keep proper history if these PRs are merged, they need to be in "rebase and merge" mode.

image

**Comment by falkTX** [comment]
That is very odd. I just cloned a brand new tenacity repo and created this PR, it should be 100% clean..

**Comment by fossdd** [comment]
In general I would suggest the team always to rebase. patches on sr.ht mailing lists should should also applied with `git am` which also rebases it.

For additional info see https://git-rebase.io/ by Drew DeVault.

**Comment by falkTX** [comment]
I can manually merge things from CLI without issues, assuming not using `--no-ff` as there is already a merge commit (needed for the README conflicts)

**Comment by falkTX** [comment]
Let me try again, I think I might have done something wrong...

**Comment by falkTX** [comment]
Recreated the repo now and force pushed, maybe good now?

File: 3RDPARTY.txt [hunk 1]
```diff
@@ -1,142 +0,0 @@
...
-MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
-CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
-NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
-STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
-ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
-ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (adrianheine) [inline 3RDPARTY.txt:142]: Why did you delete this file?

File: libraries/lib-utility/MemoryX.cpp [hunk 2]
```diff
@@ -54,5 +52,3 @@ void NonInterferingBase::operator delete(void *ptr, std::align_val_t al)
    // Call through to default operator
    ::operator delete(p);
 }
-
-#endif
```
Comment (adrianheine) [inline libraries/lib-utility/MemoryX.cpp:58]: Can't say anything about this

File: libraries/lib-utility/MemoryX.h [hunk 3]
```diff
@@ -594,10 +594,21 @@ struct UTILITY_API alignas(
...
+#if defined (_MSC_VER) && defined(_DEBUG)
+   // Versions that work in the presence of the DEBUG_NEW macro.
+   // Ignore the arguments supplied by the macro and forward to the
+   // other overloads.
+   static void *operator new(
+      std::size_t count, std::align_val_t al, int, const char *, int)
+   { return operator new(count, al); }
+   static void operator delete(
+      void *ptr, std::align_val_t al, int, const char *, int)
+   { return operator delete(ptr, al); }
```
Comment (adrianheine) [inline libraries/lib-utility/MemoryX.h:611]: Can't say anything about this either

**Review by adrianheine — COMMENTED** [review]
Looks mostly good

File: 3RDPARTY.txt (hunk 1 above)
Comment (falkTX) [reply 3RDPARTY.txt:142]: Because it generates merge conflicts even though the file is not present upstream.
Do we need it?

**Comment by falkTX** [comment]
Note that besides a little change in the README (linking to the site, similar to what audacity does) none of the changes here are from me but upstream

**Comment by emabrey** [comment]
What license are the upstream changes we are pulling?

**Comment by Semisol** [comment]
Small reminder: Please sign off your commits. Do this by:
```
git rebase --signoff HEAD~<amount of commits not signed off>
git push --force-with-lease
```
